src/ecs/entities/fog.c

```diff
@@ -151,51 +228,86 @@ void FogUpdate(Scene* scene, const usize entity)
     CPosition* position = SCENE_GET_COMPONENT_PTR(scene, position, entity);
     CKinetic* kinetic = SCENE_GET_COMPONENT_PTR(scene, kinetic, entity);
 
-    if (position->value.x < 0 && playerPosition->value.x < CTX_VIEWPORT_WIDTH)
+    const bool hasNotMoved = kinetic->velocity.x == 0;
+
+    if (hasNotMoved && playerPosition->value.x < CTX_VIEWPORT_WIDTH)
     {
-        kinetic->velocity = VECTOR2_ZERO;
         return;
     }
 
     kinetic->velocity = (Vector2)
     {
         .x = fogMoveSpeed,
-        .y = cosf(position->value.x * CTX_DT) * 32,
+        .y = cosf(ContextGetTotalTime() * 0.5f) * 16,
     };
 
-    breathingParticleSpawnTimer += CTX_DT;
     movingParticleSpawnTimer += CTX_DT;
 
-    SpawnBreathingParticles(scene, position);
-    SpawnMovingParticles(scene, position);
+    SpawnMovingParticles(scene, position, kinetic);
+
+    // Smooth phase transitioning logic for breathing.
+    {
+        static const f32 transitionIncrement = 0.1f;
+        static const f32 transitionLeeway = 0.5f;
+
+        for (usize i = 0; i < FOG_LUMP_TOTAL; ++i)
+        {
+            if (fabs(lumpRadii[i] - lumpTargetRadii[i]) <= transitionLeeway)
+            {
+                continue;
+            }
+
+            if (lumpRadii[i] < lumpTargetRadii[i])
+            {
+                lumpRadii[i] += transitionIncrement;
+            }
+            else
+            {
+                lumpRadii[i] -= transitionIncrement;
+            }
+        }
```

Comment:
I was thinking about it, and we get rid of this by using `Lerp` in `FogDraw`. There is more to it, but it would look something like this:

```c
const f32 radius = Lerp(lumpRadii[i], lumpTargetRadii[i], breathingPhaseTimer / breathingPhaseDuration) + radiusPadding;
```

I'll implement that in a future PR (along with tweaking values here and there), but for now this is fine.